lib/bundle.js

```diff
@@ -391,17 +410,26 @@ function buildWatch(builder, output, build) {
 
       building = true;
       rebuild = false;
-      if (file) 
+      if (file)
         ui.log('ok', 'File `' + path.relative(process.cwd(), file) + '` changed, rebuilding...');
       else
         ui.log('ok', 'File changes made during previous build, rebuilding...');
-      build().then(function(output) {
-        return buildWatch(builder, output, build);
-      }, function(err) {
-        ui.log('err', err.stack || err);
-        return buildWatch(builder, output, build);
-      })
-      .then(function(newWatcherRefresh) {
+
+      var buildPromise;
+      if (changeWasConfigFile) {
+        // don't bother doing a rebuild for a config file change
+        // but just keep watching so the next time a file changes
+        // the bundle is rebuilt but with the new Builder & config
+        buildPromise = buildWatch(builder, output, build);
```

Comment:
It should be possible to just do a short return on line 398 instead of this approach though.